treeherder/seta/job_priorities.py

```diff
@@ -16,67 +21,123 @@ class SetaError(Exception):
     pass
 
 
-def _process(project, build_system, job_priorities):
-    '''Return list of ref_data_name for job_priorities'''
-    jobs = []
-    # This map contains the ref_data_name of every Treeherder *test* job for this project
-    ref_data_names_map = build_ref_data_names(project, build_system)
-
-    for jp in job_priorities:
-        # if this JobPriority entry is no longer supported in SETA then ignore it
-        if not valid_platform(jp.platform):
-            continue
-        if is_job_blacklisted(jp.testtype):
-            continue
-
-        key = jp.unique_identifier()
-        if key in ref_data_names_map:
-            # e.g. desktop-test-linux64-pgo/opt-reftest-13 or builder name
-            jobs.append(ref_data_names_map[key])
+class SETAJobPriorities:
+    """
+    SETA JobPriority Implementation
+    """
+    def __init__(self):
+        self.cache_key = 'ref_data_names_cache'
+
+    def _process(self, project, build_system, job_priorities):
+        '''Return list of ref_data_name for job_priorities'''
+        jobs = []
+
+        # we cache the reference data names in order to reduce API calls
+        ref_data_names_map = cache.get(self.cache_key, 'expired')
```

Comment:
Yeah I was just going by the example in the django doc but yeah it's not necessary, I'll update thanks